components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AdditionalImagesController.java

```diff
@@ -0,0 +1,204 @@
+{
+    /** The XClass used for storing additional images. */
+    public static final EntityReference CLASS_REFERENCE = new EntityReference("ExternalImageClass", EntityType.DOCUMENT,
+        Constants.CODE_SPACE_REFERENCE);
+
+    private static final String DATA_NAME = "additional_images";
+
+    private static final String FILE_FIELD_NAME = "file";
+
+    private static final String COMMENTS_FIELD_NAME = "comments";
+
+    private static final String PRINT_FIELD_NAME = "print";
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private AttachmentAdapterFactory adapter;
+
+    @Inject
+    private Logger logger;
+
+    @Override
+    public PatientData<Attachment> load(Patient patient)
+    {
+        try {
+            XWikiDocument doc = patient.getXDocument();
+            List<BaseObject> data = doc.getXObjects(CLASS_REFERENCE);
+            if (CollectionUtils.isEmpty(data)) {
+                return null;
+            }
+
+            List<Attachment> result = new ArrayList<>(data.size());
+
+            for (BaseObject xobject : data) {
+                if (xobject == null) {
+                    continue;
+                }
+                String filename = xobject.getStringValue(FILE_FIELD_NAME);
+                XWikiAttachment xattachment = doc.getAttachment(filename);
+                if (xattachment != null) {
+                    Attachment attachment = this.adapter.fromXWikiAttachment(xattachment);
+                    attachment.addAttribute(COMMENTS_FIELD_NAME, xobject.getLargeStringValue(COMMENTS_FIELD_NAME));
+                    attachment.addAttribute(PRINT_FIELD_NAME, xobject.getIntValue(PRINT_FIELD_NAME) == 1);
+                    result.add(attachment);
+                }
+            }
+
+            return new IndexedPatientData<>(getName(), result);
+        } catch (Exception e) {
+            this.logger.error(ERROR_MESSAGE_LOAD_FAILED, e.getMessage());
+        }
+        return null;
+    }
+
+    @Override
+    public void save(Patient patient)
+    {
+        PatientData<Attachment> images = patient.getData(getName());
+        if (images == null) {
+            return;
+        }
+        try {
+            XWikiDocument doc = patient.getXDocument();
+            doc.removeXObjects(CLASS_REFERENCE);
+
+            for (Attachment image : images) {
+                BaseObject xobject = doc.newXObject(CLASS_REFERENCE, this.contextProvider.get());
+                XWikiAttachment xattachment = doc.getAttachment(image.getFilename());
+                if (xattachment == null) {
+                    xattachment = new XWikiAttachment(doc, image.getFilename());
+                    doc.addAttachment(xattachment);
+                }
+                xattachment.setContent(image.getContent());
+                DocumentReference author = image.getAuthorReference();
+                if (author != null
```

Comment:
just co confirm, the logic here is to set image author to current user if there is any author, and leave no author if there is none?